studio/components/interfaces/Organization/OAuthApps/DeleteAppModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { useParams } from 'common'
+import { useOAuthAppDeleteMutation } from 'data/oauth/oauth-app-delete-mutation'
+import { OAuthApp } from 'data/oauth/oauth-apps-query'
+import { useStore } from 'hooks'
+import { useEffect, useState } from 'react'
+import { Alert, Modal } from 'ui'
+
+export interface DeleteAppModalProps {
+  selectedApp?: OAuthApp
+  onClose: () => void
+}
+
+const DeleteAppModal = ({ selectedApp, onClose }: DeleteAppModalProps) => {
+  const { ui } = useStore()
+  const { slug } = useParams()
+  const [isDeleting, setIsDeleting] = useState(false)
+  const { mutateAsync: deleteOAuthApp } = useOAuthAppDeleteMutation()
+
+  useEffect(() => {
+    if (selectedApp) setIsDeleting(false)
+  }, [selectedApp])
```

Comment:
This can probably be removed if you put a `key` on `<DeleteAppModal />`

Reply:
can be removed after using isLoading from RQ mutation